server/src/controllers/MedicationController.js

```diff
@@ -0,0 +1,149 @@
+import MedicationModel from '../models/Medication';
+
+const MedicationController = {
+
+  /* body: medication object */
+  CreateMedication: function(req, res) {
+    MedicationModel.findOne({key: req.body.medication.key}, function(err, drug) {
+      if (drug) {
+        err = new Error(`A medication with the key ${req.body.medication.key} already exists`);
+      }
+      if (err) {
+        res.json({status: false, error: err.message});
+        return;
+      }
+      MedicationModel.create(req.body.medication, function (err) {
+        if(err) {
+          res.json({status: false, error: err.message});
+          return;
+        }
+        res.json({status: true});
+        return;
+      });
+    });
+  },
+  /* body: NA, returns: array of medication objects */
+  GetMedications: function(req, res) {
+    MedicationModel.find({drugName: req.params.name}, function(err, drugs) {
+      if (drugs.length === 0) {
+        err = new Error(`A medication with the name ${req.params.name} does not exist`);
+      }
+      if(err) {
+        res.json({status: false, error: err.message});
+        return;
+      }
+      res.json({status: true, medications: drugs});
+      return;
+    });
+  },
+
+  /* body: medication object */
+  UpdateMedication: function(req, res) {
+    MedicationModel.findOne({key: req.params.key}, function(err, drug) {
```

Comment:
To reduce code size, can we extract the code from updateMedications into a helper method, like updateMedicationsHelper() that takes in a list of medications, and then updateMedication can call updateMedicationsHelper by passing in the medication wrapped in an array?

Reply:
This wouldn't really reduce the code size since the only difference would be that most of the code gets moved to the helper. The original function would only have two lines: calling updateMedicationsHelper, and sending response object.

Reply:
Sorry I wasn't clear... there is UpdateMedication and UpdateMedications. One operates on a single medication object, the other operates on an array of medication objects. If we have UpdateMedications(array) using a helper like UpdateMedicationsHelper(array), then from UpdateMedication(medication), we can call UpdateMedicationsHelper([medication]). Does that clear it up?

Reply:
Ohh ok that makes sense. But actually, their functionalities are a bit different. UpdateMedication is used to update the fields of a single medication. So if it doesn't exist, it will cause an error (although this shouldn't happen). But UpdateMedications is used when syncing. So if a medication exists locally but not on the server, it should create a new one.

Reply:
Ah I see okay sounds good